src/Workspaces/Core/Portable/Execution/ChecksumObject.cs

```diff
@@ -73,6 +73,8 @@ private static Checksum CreateChecksum(string kind, object[] children)
     // TODO: Kind might not actually needed. see whether we can get rid of this
     internal static class WellKnownChecksumObjects
     {
+        public const string Nil = nameof(Nil);
```

Comment:
nil looks like include both null or empty or invalid. but sure can change everything to null.